client/my-sites/checkout/upsell-nudge/purchase-modal/content.tsx

```diff
@@ -1,25 +1,60 @@
 import { Button, Gridicon } from '@automattic/components';
 import { CheckoutCheckIcon, PaymentLogo } from '@automattic/composite-checkout';
+import {
+	getCreditsLineItemFromCart,
+	getItemIntroductoryOfferDisplay,
+	getSublabelAndPrice,
+} from '@automattic/wpcom-checkout';
 import { sprintf } from '@wordpress/i18n';
+import classNames from 'classnames';
 import { useTranslate } from 'i18n-calypso';
 import page from 'page';
 import { useCallback } from 'react';
 import CheckoutTerms from 'calypso/my-sites/checkout/composite-checkout/components/checkout-terms';
 import { BEFORE_SUBMIT } from './constants';
 import { formatDate } from './util';
+import type { LineItem } from '@automattic/composite-checkout';
+import type { ResponseCart, ResponseCartProduct } from '@automattic/shopping-cart';
+import type { StoredCard } from 'calypso/my-sites/checkout/composite-checkout/types/stored-cards';
+import type { MouseEventHandler, ReactNode } from 'react';
 
-function PurchaseModalStep( { children } ) {
+function PurchaseModalStep( { children, id }: { children: ReactNode; id: string } ) {
 	return (
 		<div className="purchase-modal__step">
 			<span className="purchase-modal__step-icon">
-				<CheckoutCheckIcon />
+				<CheckoutCheckIcon id={ id } />
 			</span>
 			{ children }
 		</div>
 	);
 }
 
-function OrderStep( { siteSlug, product } ) {
+function LineItemSublabelAndPrice( { product }: { product: ResponseCartProduct } ) {
```

Comment:
Fair question. I was mostly concerned with potential styling collisions further down the line. Since this component is at such a granular level, I imagined a potential future scenario where some adjustment is made to the regular checkout that has an unintended impact on the one-click checkout. Less risk for that when the "content logic" and "presentation logic" are separated in this way.

Seeing as the two instances of `LineItemSublabelAndPrice` are identical (and don't really do anything aside from calling `getSublabelAndPrice`), I paused to consider again, but I still think there's merit to this approach.

WDYT, does that make sense?

Reply:
IMHO I understand the scenario you described, and I have probably seen it in play at least once.

However, since components should be reusable by definition, potential refactors of exported components should take all existing usages into consideration during development, and then review. We can also mitigate this with tests.

My worry is that:
- Search for instances of `<LineItemSublabelAndPrice />` could become problematic as there are several instances. I know some IDEs can make the experience better.
- Inspecting code and behaviour can be confusing as they both components have identical names.

Reply:
🤔 I see your point. Thanks for your reasoning! 

That's definitely true, but I think I'd consider it a premature optimisation at this point, and one that adds additional code to this PR. The `wpcom-checkout` package already exports a number of components which are opinionated about their styling and content (the package was constructed mostly by just moving things from calypso so that they could be used by non-calypso packages) and trying to improve that is probably something better left to a larger project which examines the rest of the package's API as well.

I'd say let's wait until there's a use case where we want to change the output/styling of the component or a refactoring of the `wpcom-checkout` package exports.

Reply:
I understand your point! Went ahead with the approach suggested by @sirbrillig